code/game/objects/structures/displaycase.dm

```diff
@@ -343,3 +362,84 @@
 	name = initial(I.name)
 	icon = initial(I.icon)
 	icon_state = initial(I.icon_state)
+
+/obj/structure/displaycase/forsale
+	name = "vend-a-tray"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "laserbox0"
+	desc = "A display case with an ID-card swipe, to purchase the contents."
+	density = FALSE
+	max_integrity = 100
+	req_access = list(ACCESS_KITCHEN)
+	showpiece_type = /obj/item/reagent_containers/food
+	var/sale_price = 20
+	var/datum/bank_account/payments_acc = null
+
+/obj/structure/displaycase/forsale/update_icon()	//remind me to fix my shitcode later
+	var/icon/I
+	if(open)
+		I = icon('icons/obj/stationobjs.dmi',"laserboxb0")
+	else
+		I = icon('icons/obj/stationobjs.dmi',"laserbox0")
+	if(broken)
+		I = icon('icons/obj/stationobjs.dmi',"laserboxb0")
+	if(!showpiece && !open)
+		I = icon('icons/obj/stationobjs.dmi',"laserbox_open")
+	if(showpiece)
+		var/icon/S = getFlatIcon(showpiece)
+		S.Scale(17,17)
+		I.Blend(S,ICON_UNDERLAY,8,12)
+	src.icon = I
+	return
+
+/obj/structure/displaycase/forsale/attackby(obj/item/I, mob/living/user, params)
+	if(I.tool_behaviour == TOOL_MULTITOOL && user.a_intent == INTENT_HELP && !broken)

```

Comment:
multitool_act() is the proper standard for that tool. It activates only if you interact with it via a object with the TOOL_MULTITOOL define.